cpp/cmake/CPM.cmake

```diff
@@ -1,4 +1,4 @@
-set(CPM_DOWNLOAD_VERSION aeef56ea85f050b952623a517eb484fa6697f511)
+set(CPM_DOWNLOAD_VERSION 0.27.5)
```

Comment:
May want to use a git commit hash still, otherwise the git tag can be changed out from underneath you

Reply:
I tried that and it failed to download CPM correctly when I used the ToT hash for 0.27.5.

https://github.com/TheLartians/CPM.cmake/commit/3b404296b539e596f39421c4e92bc803b299d964

My understanding is that git tags are usually considered to be "set in stone" and won't be changed. I wasn't aware it was even possible to change them.

Reply:
> My understanding is that git tags are usually considered to be "set in stone" and won't be changed. I wasn't aware it was even possible to change them.

The way to change a git tag is to effectively delete it and then push a new tag with the same name.

What was the error when you used `3b404296b539e596f39421c4e92bc803b299d964` as the `CPM_DOWNLOAD_VERSION`?

Reply:
I imagine you'd need to change line 14 to: `https://raw.githubusercontent.com/TheLartians/CPM.cmake/${CPM_DOWNLOAD_VERSION}/cmake/CPM.cmake`

Reply:
Okay that works, but is this a real concern? e.g., we just use the versions for both GTest and GBench instead of commit hashes.

Reply:
Using a git commit hash ensures the code never changes out from underneath us. The concern for CPM is it's effectively a single maintainer where if they / the repo get compromised we could be exposed. A bit less of a concern for Google run repos, but still possible.